features/step_definitions/profile_steps.rb

```diff
@@ -8,6 +8,6 @@
 
 When(/^I delete a photo$/) do
   find('.photo.loaded').hover
-  find('.delete', :match => :first).click
+  find('.delete', :visible => false, :match => :first).click
```

Comment:
The button has to be visible for the user so looking for an invisible button doesn't look like the right thing to do.

Here is another solution which fixed the test for me:
```diff
diff --git a/features/step_definitions/profile_steps.rb b/features/step_definitions/profile_steps.rb
index 70f9acc..60fd76f 100644
--- a/features/step_definitions/profile_steps.rb
+++ b/features/step_definitions/profile_steps.rb
@@ -7,7 +7,7 @@ And /^I mark myself as safe for work$/ do
 end
 
 When(/^I delete a photo$/) do
-  find('.photo.loaded').hover
-  find('.delete', :visible => false, :match => :first).click
+  find('.photo.loaded .thumbnail', :match => :first).hover
+  find('.delete', :match => :first).click
 end
```